algorithms/polish/LICENSE

```diff
@@ -0,0 +1,21 @@
+MIT License
```

Comment:
Nothing seems to actually reference this file, and the PR against the snowball repo doesn't mention this licence.

The Snowball compiler and other algorithms are all released under the 3-clause BSD licence, which is also a permissive licence and really pretty similar in practical terms (it does have an additional "no endorsements" clause, but that shouldn't make a difference to any reasonable use).

Are you happy to just license under 3-clause BSD?  I'm not keen to have licence proliferation within the Snowball distribution.